google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/StatementExecutor.java

```diff
@@ -129,12 +141,13 @@ public boolean hasDuration() {
           .build();
 
   /** Creates an {@link ExecutorService} for a {@link StatementExecutor}. */
-  private static ExecutorService createExecutorService() {
-    return new ThreadPoolExecutor(
-        1, 1, 0L, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<Runnable>(), THREAD_FACTORY);
+  private static ListeningExecutorService createExecutorService() {
```

Comment:
Create a `ListeningExecutorService`. The `ListenableFuture`s that are returned by this can be converted to `ApiFuture`s.